### PR TITLE
Feature/kuberhealthy pod annotations

### LINKED ISSUE #115: Add podAnnotations section to helm chart
If a podAnnotations section is added to the helm chart, it is possible to add annotations to kuberhealthy which can be used for setting up external prometheus scrapers (eg datadog).

Eg:
```
  podAnnotations:
    ad.datadoghq.com/kuberhealthy.check_names: '["prometheus"]'
    ad.datadoghq.com/kuberhealthy.init_configs: '[{}]'
    ad.datadoghq.com/kuberhealthy.instances: '[{"prometheus_url":
      "http://%%host%%:8080/metrics", "namespace": "kuberhealthy", "metrics":
      ["kuberhealthy_*"]}]'
    prometheus.io/port: "8080"
    prometheus.io/scrape: "true"
```

### PR DESCRIPTION
**Pull request by logicfox**
#### What this PR does / why we need it:
Allow addition of pod annotations to the deployment template. This is needed in certain use-cases like Datadog monitoring, which requires annotations on the pod to scrape Prometheus metrics.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes [#115](https://github.com/Comcast/kuberhealthy/issues/115)

#### Special notes for your reviewer:
@integrii @lolimjake @ihoegen This could additionally address https://github.com/Comcast/kuberhealthy/issues/119

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/kuberhealthy/templates/deployment.yaml [hunk 1]
```diff
@@ -20,6 +20,12 @@ spec:
     type: RollingUpdate
   template:
     metadata:
+      {{- if .Values.deployment.podAnnotations }}
+      annotations:
+      {{- range $key, $value := .Values.deployment.podAnnotations }}
```
Comment (ihoegen) [inline stable/kuberhealthy/templates/deployment.yaml:25]: You should just use the toYaml function in helm

**Review by ihoegen — CHANGES_REQUESTED** [review]

File: stable/kuberhealthy/templates/deployment.yaml (hunk 1 above)
Comment (davidkarlsen) [reply stable/kuberhealthy/templates/deployment.yaml:25]: iterating allows to add the quote function - so I think it is OK

**Pull request merged**